File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        window = set()
        left = 0
        longest = 0
        for end in range(len(s)):
            while s[end] in window:
                window.remove(s[left])
                left += 1
            
            window.add(s[end])
            longest = max(longest, len(window))
        
        return longest

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var window: Set<Character> = Set()
        var longest = 0

        let array = Array(s)
        var left = 0

        for right in 0..<array.count {
            while window.contains(array[right]) {
                window.remove(array[left])
                left += 1
            }

            window.insert(array[right])

            longest = max(longest, window.count)
        }

        return longest
    }
}